cmd/prometheus/main.go

```diff
@@ -237,6 +239,9 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 			case "delayed-compaction":
 				c.tsdb.EnableDelayedCompaction = true
 				level.Info(logger).Log("msg", "Experimental delayed compaction is enabled.")
+			case "promql-delayed-name-removal":
+				c.promqlEnableDelayedNameRemoval = true
+				level.Info(logger).Log("msg", "PromQL delayed name removal enabled.")
```

Comment:
Nit: Let's start the message with "Experimental" (as most of the other messages for feature flags).